### PR TITLE
Upgrade supported PostgreSQL to version 16

### LINKED ISSUE #29875: Upgrade supported PostgreSQL to version 16
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

The PostgreSQL support should be upgraded to version 16.

### Version

latest

### Regression

- [ ] The issue is a regression

### Expected behavior

-

### Actual behavior

-

### How to Reproduce?

-

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by miquelsi**
Closes #29875

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -157,7 +157,7 @@
         <!-- Databases - also published to db.adoc as "Tested Version" -->
         <mysql.version>8.0</mysql.version>
         <mysql-jdbc.version>8.3.0</mysql-jdbc.version>
-        <postgresql.version>15</postgresql.version>
+        <postgresql.version>16</postgresql.version>
         <aurora-postgresql.version>15.5</aurora-postgresql.version>
```
Comment (ahus1) [inline pom.xml:161]:
```suggestion
        <aurora-postgresql.version>16.1</aurora-postgresql.version>
```

**Review by ahus1 — CHANGES_REQUESTED** [review]
When upgrading regular PostgreSQL to 16, please also upgrade Aurora PostgreSQL. The version of this flavor is always specified as major-minor, therefore it has a separate entry.

**Comment by stianst** [comment]
We should probably also updated the tested version in https://www.keycloak.org/server/db?

**Comment by ahus1** [comment]
@stianst - the tested version in the docs is extracted from the `pom.xml` - so it will always be up to date.

**Review by ahus1 — DISMISSED** [review]

**Comment by ahus1** [comment]
@miquelsi - is that something you want to put into the release notes or migration guide?

**Comment by miquelsi** [comment]
> @miquelsi - is that something you want to put into the release notes or migration guide?

Yes, in the release notes

**Review by ahus1 — COMMENTED** [review]
@miquelsi - I don't see a release note change in this PR you mentioned in your previous comment. Did I miss something?

File: docs/documentation/release_notes/topics/25_0_0.adoc [hunk 2]
```diff
@@ -183,4 +183,8 @@ For more details, see the link:{upgradingguide_link}[{upgradingguide_name}].
 As a consequence of the removal of deprecated methods from `AccessToken`, `IDToken`, and `JsonWebToken`,
 the `SingleUseObjectKeyModel` also changed to keep consistency with the method names related to expiration values.
 
-For more details, see the link:{upgradingguide_link}[{upgradingguide_name}].
\ No newline at end of file
+For more details, see the link:{upgradingguide_link}[{upgradingguide_name}].
+
+= Support for PostgreSQL 16
+
+The supported and tested databases now include PostgreSQL 16
```
Comment (ahus1) [inline docs/documentation/release_notes/topics/25_0_0.adoc:190]:
```suggestion
The supported and tested databases now include PostgreSQL 16.
```

**Review by ahus1 — APPROVED** [review]

**Pull request merged**